src/html5_parser/node.rs

```diff
@@ -447,4 +547,181 @@ mod test {
         let node = Node::new_document();
         assert_eq!(node.data, NodeData::Document {});
     }
+
+    #[test]
+    fn test_contains_attribute_non_element() {
+        let node = Node::new_document();
+        let result = node.contains_attribute("x".to_string());
+        match result {
+            Err(_) => assert!(true),
+            Ok(_) => assert!(false),
+        }
+    }
+
+    #[test]
+    fn test_contains_attribute() {
+        let mut attr = HashMap::new();
+        attr.insert("x".to_string(), "value".to_string());
+
+        let node = Node::new_element("node", attr.clone(), HTML_NAMESPACE);
+
+        match node.contains_attribute("x".to_string()) {
+            Err(_) => assert!(false),
+            Ok(result) => {
+                assert_eq!(result, true);
+            }
+        }
+
+        match node.contains_attribute("z".to_string()) {
+            Err(_) => assert!(false),
+            Ok(result) => {
+                assert_eq!(result, false);
+            }
+        }
+    }
+
+    #[test]
+    fn test_insert_attrubte_non_element() {
+        let mut node = Node::new_document();
+        let result = node.insert_attribute("name".to_string(), "value".to_string());
+        match result {
+            Err(_) => assert!(true),
+            Ok(_) => assert!(false),
```

Comment:
For this and similar tests, see above.

Reply:
thanks, didn't know this shorthand, I'll modify the others

Reply:
updated all others in [a378069](https://github.com/jaytaph/gosub-browser/pull/70/commits/a378069a9588ccc4f216c90ad211a32065133950)

Reply:
No worries at all.  I'm still learning a lot as I go, myself.

Reply:
yeah, I've used Rust for about 3 days before stumbling across this project lol but it's been a decent transition since I come from the C world so the learning curve hasn't been terrible (yet) - although there's still many many things to learn